### PR TITLE
Sending outage notifications to responder phone numbers with instructions for power-cycling hubs (CU-faxfbg, CU-vf8x83, CU-13kqxyt)

### PR DESCRIPTION
**Pull request by mariocimet**
## Test Plan

- :heavy_check_mark:Log onto chatbot dashboard and verify that timestamps (Created At, Updated At, Responded At) are displayed appropriately and in Vancouver time
- :heavy_check_mark: The heartbeat dashboard displays the `location_description` for each hub
- :heavy_check_mark: Create a test hub on `chatbot-dev` with my own number as responder phone and two twilio numbers as heartbeat recipients
   - :heavy_check_mark: Upon initial disconnection
       - :heavy_check_mark:  Sentry disconnection messages are only once for each threshold (ping, flic, and heartbeat) passed
       - :heavy_check_mark: Initial disconnection text message is sent to me and both Twilio numbers after the external threshold is passed
   - :heavy_check_mark: After every time the subsequent alert threshold is passed
      - :heavy_check_mark: No Sentry messages
      - :heavy_check_mark: Disconnection reminder message is sent to me and both Twilio numbers each time we pass the subsequent alert threshold
   - :heavy_check_mark: After reconnecting the hub (did this by setting the `last_seen_*` values in the DB to `now()` )
      - :heavy_check_mark:  Reconnection is sent to me and both Twilio number
      - :heavy_check_mark:  Sentry reconnection messages are only once for each threshold now reconnected 
   - :heavy_check_mark: If the hub is muted
      - :heavy_check_mark: No disconnection, reconnection, or reminder messages are sent when the hub is muted
- :heavy_check_mark: Verify that the `location_description` text makes sense for all our production hubs
- :heavy_check_mark: Sent heartbeat message to `/heartbeat` and see that it reconnects the hub
- :heavy_check_mark: Run `add_hubs.sh` and see that new hubs are added to the DB
- :heavy_check_mark: Run the `setup_server.sh` script and see that the server is down when the DB updates error out after adding the `installation_id` and `location_description` columns to the `hubs` table.
- :heavy_check_mark: Run the `setup_server.sh` script and see that it completes successfully and the server is back up after adding in non-null values for the `installation_id` and `location_description` in the `hubs` table.
- :heavy_check_mark" Run the `setup_server.sh` script with valid and invalid values being sent into the `certbot` command to see that it is correctly getting "1" as user input when it runs

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -492,47 +494,47 @@ async function checkHeartbeat() {
...
+      const flicLastSeenTime = hub.flicLastSeenTime
+      const flicDelayMillis = currentTime - flicLastSeenTime
+      const heartbeatLastSeenTime = hub.heartbeatLastSeenTime
+      const heartbeatDelayMillis = currentTime - heartbeatLastSeenTime
+      const pingLastSeenTime = hub.flicLastPingTime
+      const pingDelayMillis = currentTime - pingLastSeenTime
+
+      const flicHeartbeatExceeded = flicDelayMillis > helpers.getEnvVar('HEARTBEAT_THRESHOLD_SECONDS')
+      const pingHeartbeatExceeded = pingDelayMillis > helpers.getEnvVar('HEARTBEAT_THRESHOLD_SECONDS')
+      const piHeartbeatExceeded = heartbeatDelayMillis > helpers.getEnvVar('HEARTBEAT_THRESHOLD_SECONDS')
```
Comment (mariocimet) [inline server/server.js:508]: Different thresholds for us to get sentry alerts vs clients getting a text message

File: server/.env.example [hunk 2]
```diff
@@ -84,3 +84,6 @@ ONESIGNAL_APP_ID_TEST=guid-here
 # OneSignal App ID from OneSignal --> Settings --> Keys & IDs --> REST API Key
 ONESIGNAL_API_KEY=jumbleOfCharactersHere
 ONESIGNAL_API_KEY_TEST=jumbleOfCharactersHere
+
+# Heartbeat Threshold
+HEARTBEAT_THRESHOLD=900000
```
Comment (trisapeace) [inline server/.env.example:89]: I think we were trying to move to having all time things in seconds, right?

File: server/db/017-add-installation-foreign-key-to-hubs.sql [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    -- The migration ID of this file
+    migrationId := 17;
+
+    -- Get the migration ID of the last file to be successfully run
+    SELECT MAX(id) INTO lastSuccessfulMigrationId
+    FROM migrations;
+
+    -- Only execute this script if its migration ID is next after the last successful migration ID
+    IF migrationId - lastSuccessfulMigrationId = 1 THEN
+        ALTER TABLE hubs ADD COLUMN installation_id uuid REFERENCES installations (id);
```
Comment (trisapeace) [inline server/db/017-add-installation-foreign-key-to-hubs.sql:14]: Does this need a `NOT NULL` added to it?

File: server/db/db.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 const { CHATBOT_STATE, helpers } = require('brave-alert-lib')
+const { runQuery } = require('brave-alert-lib/lib/helpers')
```
Comment (trisapeace) [inline server/db/db.js:2]: `helpers` has already been required above. So you can just use `helpers.runQuery` anywhere in the file.

File: server/server.js [hunk 5]
```diff
@@ -29,7 +25,7 @@ const unrespondedSessionAlertTimeoutMillis = helpers.getEnvVar('FALLBACK_TIMEOUT
 const heartbeatDashboardTemplate = fs.readFileSync(`${__dirname}/heartbeatDashboard.mst`, 'utf-8')
 const chatbotDashboardTemplate = fs.readFileSync(`${__dirname}/chatbotDashboard.mst`, 'utf-8')
 
-app.use(bodyParser.urlencoded({ extended: true }))
+app.use(express.urlencoded({ extended: true }))
```
Comment (trisapeace) [inline server/server.js:28]: Can `body-parser` be removed from package.json and package-lock.json now?

File: server/db/017-add-installation-foreign-key-to-hubs.sql (hunk 3 above)
Comment (mariocimet) [reply server/db/017-add-installation-foreign-key-to-hubs.sql:14]: in a subsequent script? there's already things in this table and I forgot how we've dealt with that. I'm thinking run this one, update values, and then set the constraint after?

Comment (trisapeace) [reply server/db/017-add-installation-foreign-key-to-hubs.sql]: Good point!

Comment (trisapeace) [reply server/db/017-add-installation-foreign-key-to-hubs.sql]: I guess you could do it all in one script here by hardcoding the IDs that you want and then setting it to `NOT NULL` . But it seems cludgey. Maybe a subsequent pull request with nothing but the `NOT NULL` change that can be merged after the next deployment?

File: server/package.json [hunk 6]
```diff
@@ -5,17 +5,18 @@
   "main": "server.js",
   "dependencies": {
     "body-parser": "^1.18.3",
-    "brave-alert-lib": "https://github.com/bravetechnologycoop/brave-alert-lib#v6.1.0",
```
Comment (trisapeace) [inline server/package.json:8]: Note to self: update to real version number after merging https://github.com/bravetechnologycoop/brave-alert-lib/pull/32

File: CHANGELOG.md [hunk 7]
```diff
@@ -9,24 +9,34 @@ Please note that the date associated with a release is the date the code
...
+
+### Fixed
+
+- Documentation in `add_hubs.sh` script
 
 ## [6.1.0] - 2021-11-29
 
 ### Changed
 
-- `setup_pi.yaml` no longer triggers `apt update` using the apt module since ansible doesn't support allowing releaseinfo changes.
```
Comment (trisapeace) [inline CHANGELOG.md:29]: This was included in `master` through some bad merge, I think.

**Comment by trisapeace** [comment]
Task linked: [CU-13kqxyt replace deprecated express body-parser usage in Buttons, Sensors, and brave-alert-lib](https://app.clickup.com/t/13kqxyt)

**Comment by trisapeace** [comment]
Task linked: [CU-vf8x83 send Button Hub outage alerts directly to clients with instructions on how to fix problems](https://app.clickup.com/t/vf8x83)

**Comment by trisapeace** [comment]
Task linked: [CU-faxfbg fix builtin documentation in add_hubs.sh](https://app.clickup.com/t/faxfbg)

File: server/db/017-add-installation-foreign-key-to-hubs.sql (hunk 3 above)
Comment (trisapeace) [reply server/db/017-add-installation-foreign-key-to-hubs.sql:14]: Subsequent pull request here so we don't forget to do this: https://github.com/bravetechnologycoop/BraveButtons/pull/136

File: server/package.json (hunk 6 above)
Comment (trisapeace) [reply server/package.json:8]: Done.

**Comment by trisapeace** [comment]
OK, I think I'm happy with this one. Very cool. I'm looking forward to it working.

**Pull request merged**